### PR TITLE
feat(nft): fetch for all supported networks

### PR DESCRIPTION
**Pull request by satish-ravi**
### Description

Fetch nfts for all supported network ids from [dynamic config](https://console.statsig.com/4plizaPmWwPL21ASV4QAO0/dynamic_configs/multi_chain_features/preview/6DUt41gm1Tl6mQbi3LWqm)

### Test plan

Unit tests, manually by using an account with NFTs (thanks @MuckT )

https://github.com/valora-inc/wallet/assets/5062591/cbea165c-bb09-4c1d-ade5-2f8f92adec12

### Related issues

- Fixes ACT-1017

### Backwards compatibility

Yes

### CODE REVIEW

File: src/redux/migrations.ts [hunk 1]
```diff
@@ -1507,4 +1507,11 @@ export const migrations = {
       ),
     },
   }),
+  181: (state: any) => ({
+    ...state,
+    nfts: {
+      ...state.nfts,
+      nfts: [],
```
Comment (satish-ravi) [inline src/redux/migrations.ts:1514]: we could update existing nfts in place, but we refetch all nfts on app start, so should be fine to just clean this

File: src/nfts/NftGallery.tsx
Comment (MuckT) [inline src/nfts/NftGallery.tsx:1]: Unrelated, but we should make a ticket to remove this file.

File: src/nfts/saga.test.tsx [hunk 2]
```diff
@@ -25,26 +28,42 @@ describe('Given Nfts saga', () => {
...
         )
         .run()
 
       expect(mockFetch).toHaveBeenCalledWith(
-        'https://api.alfajores.valora.xyz/getNfts?address=0xabc',
+        'https://api.alfajores.valora.xyz/getNfts?address=0xabc&networkId=celo-alfajores',
         {
           headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
           method: 'GET',
+          signal: expect.any(AbortSignal),
```
Comment (MuckT) [inline src/nfts/saga.test.tsx:58]: This AbortSignal looks new was it recently added in the backend?

File: src/nfts/saga.ts [hunk 3]
```diff
@@ -11,6 +14,37 @@ import { call, put, select, spawn, takeLeading } from 'typed-redux-saga'
 
 const TAG = 'NftsSaga'
 
+export async function fetchNftsForSupportedNetworks(address: string): Promise<NftWithNetworkId[]> {
```
Comment (MuckT) [inline src/nfts/saga.ts:17]: Should this function be tested independently with unit tests or are the tests in the saga sufficent?

File: src/nfts/types.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import { NetworkId } from 'src/transactions/types'
```
Comment (MuckT) [inline src/nfts/types.ts:1]: It's a bit strange that networkId lives in transaction types, when it's widely shared. Maybe we should have a src/shared/types.ts, but that's outside of the scope of this PR.

File: src/statsig/constants.ts [hunk 5]
```diff
@@ -81,6 +81,7 @@ export const DynamicConfigs = {
       showTransfers: [networkConfig.defaultNetworkId],
       showWalletConnect: [networkConfig.defaultNetworkId],
       showApprovalTxsInHomefeed: [],
+      showNfts: [networkConfig.defaultNetworkId],
```
Comment (MuckT) [inline src/statsig/constants.ts:84]: Is this feature flag usage supposed to be included in this PR?

File: src/nfts/saga.ts (hunk 3 above)
Comment (satish-ravi) [reply src/nfts/saga.ts:17]: this is covered in the saga test, just a convenience function. Will drop the export

File: src/statsig/constants.ts (hunk 5 above)
Comment (satish-ravi) [reply src/statsig/constants.ts:84]: this is not a feature flag, just a new property on the existing multi_chain_features dynamic config

Comment (MuckT) [reply src/statsig/constants.ts:84]: Approved the changes to `multi_chain_features` in Statsig.

**Review by MuckT — APPROVED** [review]

**Comment by silasbw** [comment]
Rad :guitar:

**Pull request merged**